build_js.sh

```diff
@@ -20,7 +20,7 @@ mkdir -p build_js
 cd build_js
 
 cmake ../src \
-    -DCORRADE_RC_EXECUTABLE=../build_corrade-rc/deps/corrade/src/Corrade/Utility/corrade-rc \
+    -DCORRADE_RC_EXECUTABLE=../build_corrade-rc/RelWithDebInfo/bin/corrade-rc \
```

Comment:
This seems unrelated and assumes the build-type of corrade-rc.

Reply:
With 2019.10 magnum is by default putting all binaries and libraries into a single location to make it easier to load dynamic plugins from the build dir (and also avoid DLL hell for Windows users). So this is a direct result of that. I could override that with `CMAKE_RUNTIME_OUTPUT_LOCATION` for the CI script if desired.

Reply:
Interesting, so binaries are no longer being built in the usual location of CMAKE_CURRENT_BINARY_DIR?

Reply:
Not, unless you set one of the `CMAKE_{RUNTIME,LIBRARY,ARCHIVE}_OUTPUT_LOCATION`, in which case that setting is preserved. This also affects only Magnum used as a subproject, in case of a system-wide installed Magnum it's assumed PATH is set up properly and plugins are in expected locations, so there's less reasons to mess with this. [Docs for reference.](https://doc.magnum.graphics/magnum/cmake.html#cmake-subproject)

This makes the workflow easier for majority of users and so it's enabled by default. Habitat builds everything including plugins as static, so there it doesn't make much difference (although once I got used to binaries being all in one place, I have to admit it's much more convenient). If you wish to preserve the original behavior, I can override this.

Reply:
I thought release was the default build. Maybe set CMAKE_BUILD_TYPE explicitly above when corrade-rc is built. That way, we aren't relying on a default.

Reply:
`RelWithDebInfo` is set as default [here in CMakeLists.txt](https://github.com/facebookresearch/habitat-sim/blob/5ac8a748e1054d9ab40c820d5ea9ca0b9452a22b/src/CMakeLists.txt#L40-L44) and [here as the opposite of Debug in setup.py](https://github.com/facebookresearch/habitat-sim/blob/5ac8a748e1054d9ab40c820d5ea9ca0b9452a22b/setup.py#L223). It was before my time so I don't know the reasons behind, but I'd say it's a pretty strong default :)

(AFAIK, CMake's default build type is an empty string, so neither Debug, Release, nor RelWithDebInfo.)

Reply:
Ah.